examples/producer/src/main/java/io/strimzi/examples/producer/ExampleProducer.java

```diff
@@ -0,0 +1,107 @@
+ */
+package io.strimzi.examples.producer;
+
+import io.strimzi.kafka.oauth.client.ClientConfig;
+import io.strimzi.kafka.oauth.common.Config;
+import io.strimzi.kafka.oauth.common.ConfigProperties;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.StringSerializer;
+
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+
+public class ExampleProducer {
+
+    public static void main(String[] args) {
+
+        String topic = "Topic1";
+
+        Properties defaults = new Properties();
+        Config external = new Config();
+
+        // TODO: Set KEYCLOAK_IP to be able to connect to Keycloak
+        //  Use 'keycloak.ip' system property or KEYCLOAK_IP env variable
+
+        final String KEYCLOAK_IP = external.getValue("keycloak.ip", "keycloak");
+        final String REALM = external.getValue("realm", "demo");
+        final String TOKEN_ENDPOINT_URI = "http://" + KEYCLOAK_IP+ ":8080/auth/realms/" + REALM + "/protocol/openid-connect/token";
+
+        // You can also configure token endpoint uri directly via 'oauth.token.endpoint.uri' system property
+        //  or OAUTH_TOKEN_ENDPOINT_URI env variable
+        defaults.setProperty(ClientConfig.OAUTH_TOKEN_ENDPOINT_URI, TOKEN_ENDPOINT_URI);
+
+
+        // TODO: By defaut this client uses preconfigured clientId and secret to authenticate.
+        //  You can set OAUTH_ACCESS_TOKEN or OAUTH_REFRESH_TOKEN to override default authentication.
+        //
+        //  If access token is configured, it is passed directly to Kafka broker
+        //  If refresh token is configured, it is used in conjunction with clientId and secret
+        //
+        //  See README.md for more info.
+
+        final String ACCESS_TOKEN = external.getValue(ClientConfig.OAUTH_ACCESS_TOKEN, null);
+
+        if (ACCESS_TOKEN == null) {
+            defaults.setProperty(Config.OAUTH_CLIENT_ID, "kafka-producer-client");
+            defaults.setProperty(Config.OAUTH_CLIENT_SECRET, "kafka-producer-client-secret");
+        }
+
+        // Use 'preferred_username' rather than 'sub' for principal name
+        defaults.setProperty(Config.OAUTH_USERNAME_CLAIM, "preferred_username");
+
+        // Resolve external configurations falling back to provided defaults
+        ConfigProperties.resolveAndExportToSystemProperties(defaults);
+
+
+        Properties props = buildProducerConfig();
+        Producer<String, String> producer = new KafkaProducer<>(props);
+
+        for (int i = 0; ; i++) {
+            try {
+
+                producer.send(new ProducerRecord<>(topic, "Message " + i))
+                        .get();
+
+                System.out.println("Produced Message " + i);
+
+
```

Comment:
tbh, I also like blank lines for readability but two blank lines in this case are too much :-)

Reply:
Do we have a style guide that covers empty lines? Otherwise it';s purely a matter of taste. I don't much care if it's one, two, three. If one is more readable to others than two, then sure, there can be one - it's just empty lines.

Reply:
We don't have a style guide. Personally I wouldn't want one: I would hope we can find a consensus which works for all of us without needing a style guide (which then needs to be enforced if it's to have any value). My personal preference is to use single blank lines sparingly. I find I fit more code on the screen that way. My main desire is to avoid a mixture of styles in a given file (and ideally in a given repo).